### PR TITLE
Update graylog

### PR DESCRIPTION
**Pull request by bbenno**
- Update Graylog 6.0
- Remove Graylog 5.1 (due to EOL)
- In NixOS set default version to 6.0 (prev. was now removed version 5.1)

[Migration Guide 5.1 → 5.2](https://go2docs.graylog.org/5-2/upgrading_graylog/upgrading_to_graylog_5.2.x.htm)

[Migration Guide 5.2 → 6.0](https://go2docs.graylog.org/6-0/upgrading_graylog/upgrading_to_graylog_6.0.x.htm)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/logging/graylog.nix [hunk 1]
```diff
@@ -35,8 +35,8 @@ in
 
       package = lib.mkOption {
         type = lib.types.package;
-        default = if lib.versionOlder config.system.stateVersion "23.05" then pkgs.graylog-3_3 else pkgs.graylog-5_1;
-        defaultText = lib.literalExpression (if lib.versionOlder config.system.stateVersion "23.05" then "pkgs.graylog-3_3" else "pkgs.graylog-5_1");
+        default = pkgs.graylog-6_0;
+        defaultText = lib.literalExpression "pkgs.graylog-6_0";
```
Comment (leona-ya) [inline nixos/modules/services/logging/graylog.nix:39]: I'm not so happy with silently changing the default for all state versions. It's probably the easiest to add an assertion for old state versions like in https://github.com/NixOS/nixpkgs/pull/345625/files#diff-5834a9cebfb2c9d2d6863e68be9c225f325f77d1c5a1a5e55f66fa3298995a93R90 or only set the default when the state version is >=25.05

**Review by leona-ya — CHANGES_REQUESTED** [review]
Generally I'm okay with removing the package, but not with the package default change.

**Comment by bbenno** [comment]
> Generally I'm okay with removing the package, but not with the package default change.

I'll split the PR up into dedicated updates for each version (that can then be backported on an individual level) and another for the removal of Graylog 5.1.

File: nixos/modules/services/logging/graylog.nix (hunk 1 above)
Comment (bbenno) [reply nixos/modules/services/logging/graylog.nix:39]: Thank you very much for pointing out the assertion pattern that I was not aware of.

However, after grep'ing for this pattern and recognizing that it is only used for UniFi, I decided to go with the pattern PostgreSQL uses:

in the options part
https://github.com/NixOS/nixpkgs/blob/924dd674814aaf618e329dd252e07f9a083c5f20/nixos/modules/services/databases/postgresql.nix#L89-L91

In the implementation part (`config = mkIf cfg.enable`)
https://github.com/NixOS/nixpkgs/blob/924dd674814aaf618e329dd252e07f9a083c5f20/nixos/modules/services/databases/postgresql.nix#L491-L512

**Pull request closed without merging**